### PR TITLE
feat: navigation with dropdowns

### PR DESCRIPTION
**Pull request by AmandaKr**
## Description 📝

This PR introduces the capability to display dropdowns in the Navigation.

## Related Issues 🔗

JIRA issue: [FSU-290](https://klau.atlassian.net/browse/FSU-290)

## Preview Links 🔍

- [Navigation](https://design-system-git-fsu-290-sterczace-uszy.vercel.app/?path=/story/blocks-navigation--controls)

## Ask for Help 🙋‍♀️🙋‍♂️
I don't know why cypress tests are failing 😭

### CODE REVIEW

**Review by paulgrym — COMMENTED** [review]
Looks great 🚀 I notice one problem - after expanding the last navigation element, not all elements are fully visible:
image

**Comment by AmandaKr** [comment]
> Looks great 🚀 I notice one problem - after expanding the last navigation element, not all elements are fully visible: image PS e2e test are failing

I think it's because the navigation doesn't fit in this storybook container, try opening the navigation in a separate tab for the whole screen

**Comment by paulgrym** [comment]
> > Looks great 🚀 I notice one problem - after expanding the last navigation element, not all elements are fully visible: image PS e2e test are failing
> 
> I think it's because the navigation doesn't fit in this storybook container, try opening the navigation in a separate tab for the whole screen

In full screen on my laptop the text is also cut off. I think the dropdown should pin to the right when it is not fully visible

**Review by imklau — COMMENTED** [review]
The navigation looks awesome now! @AmandaKr, great work! 🎉 
And what's more important, the current nav still works :D

I've just asked Dorota about the dropdowns because I feel like the box shadow is missing, and their alignment could be a bit different 🤔

**Review by paulgrym — APPROVED** [review]
Excellent🌟

**Comment by imklau** [comment]
@AmandaKr It's quite difficult to hover over the sublinks 🙈 


https://github.com/non-profit-dev/sterczace-uszy/assets/23500678/cc659d80-b2b5-4947-b1d4-fa73b5a75e5b

File: packages/design-system/blocks/navigation/Navigation.jsx [hunk 1]
```diff
@@ -76,4 +160,12 @@ const Navigation = () => {
   )
 }
 
+Navigation.propTypes = {
+  navigationData: string,
```
Comment (imklau) [inline packages/design-system/blocks/navigation/Navigation.jsx:164]: I think this `data` prop should be different 🤔

**Comment by imklau** [comment]
@AmandaKr I've just added a `DropdownWrapper` element with absolute position and top padding (but without any other styles) which seems to resolve the issue with empty space between the menu link and the dropdown.

The `DropdownWrapper` isn't visible for the users, but it's positioned right below the Menu Link. Its child element has this white background and box shadow.

Feel free to review my changes and let me know what you think! 😄

**Comment by AmandaKr** [comment]
> @AmandaKr I've just added a `DropdownWrapper` element with absolute position and top padding (but without any other styles) which seems to resolve the issue with empty space between the menu link and the dropdown.
> 
> The `DropdownWrapper` isn't visible for the users, but it's positioned right below the Menu Link. Its child element has this white background and box shadow.
> 
> Feel free to review my changes and let me know what you think! 😄

@imklau   You're right, your solution is good, I didn't notice it before, thanks!

**Review by imklau — APPROVED** [review]
Really great work! 👏🏼

**Pull request merged**